src/ploigos_step_runner/step_implementers/audit_attestation/opa.py

```diff
@@ -0,0 +1,184 @@
+        config,
+        environment=None
+    ):
+
+        super().__init__(
+            workflow_result=workflow_result,
+            parent_work_dir_path=parent_work_dir_path,
+            config=config,
+            environment=environment
+        )
+
+    @staticmethod
+    def step_implementer_config_defaults():
+        """Getter for the StepImplementer's configuration defaults.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+
+        """
+        return DEFAULT_CONFIG
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    def _audit_attestation(self, # pylint: disable=no-self-use
+        workflow_attestation_file_path,
+        workflow_policy_file_path,
+        workflow_policy_query):
+
+        opa_attestation_stdout_result = StringIO()
+        opa_attestation_stdout_callback = create_sh_redirect_to_multiple_streams_fn_callback([
+            sys.stdout,
+            opa_attestation_stdout_result
+        ])
+        sh.opa( # pylint: disable=no-member
+            'opa',
+            'eval',
+            '--fail',
+            '--entry',
+            '-i',
+            workflow_attestation_file_path,
+            '-d',
+            workflow_policy_file_path,
+            workflow_policy_query,
+            _out=opa_attestation_stdout_callback,
+            _err_to_out=True,
+            _tee='out'
+        )
+
+        return_code = sh.ErrorReturnCode
+
+        return opa_attestation_stdout_callback, return_code
+
+
+
+
+
+    def _run_step(self):
+        """Runs the step implemented by this StepImplementer.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        step_result = StepResult.from_step_implementer(self)
+
+        work_dir = self.work_dir_path
+
+        uri_artifact_key = 'evidence-uri'
+
+        #workflow attestation uri
+        workflow_attestation_uri = self.get_value(uri_artifact_key)
+
+        if workflow_attestation_uri is None:
+            step_result.success = False
+            step_result.message = "No value found for " + uri_artifact_key
+            return step_result
+
+        workflow_attestation_file_path = download_source_to_destination(
+            workflow_attestation_uri,
+            work_dir
+        )
```

Comment:
Consider breaking `workflow attestation uri` into a separate method for easier testing.

Reply:
Do you mean lines 142-147?

Reply:
That could probably go into its own method.